src/Microsoft.DotNet.ScenarioTests.SdkTemplateTests/SdkTemplateTest.cs

```diff
@@ -23,16 +23,27 @@ public SdkTemplateTest(string scenarioName, DotNetLanguage language, string targ
         TargetRid = targetRid;
     }
 
-    internal void Execute(DotNetSdkHelper dotNetHelper, string testRoot, string[]? frameworks = null)
+    internal void Execute(DotNetSdkHelper dotNetHelper, string testRoot, string[]? frameworks = null, string? PreMadeSolution = null)
     {
         // Don't use the cli language name in the project name because it may contain '#': https://github.com/dotnet/roslyn/issues/51692
         string projectName = $"{ScenarioName}_{Template}_{Language}";
         string customNewArgs = Template.IsAspNetCore() && NoHttps ? "--no-https" : string.Empty;
         string projectDirectory = Path.Combine(testRoot, projectName);
 
-        Directory.CreateDirectory(projectDirectory);
-
-        dotNetHelper.ExecuteNew(Template.GetName(), projectName, projectDirectory, Language.ToCliName(), customArgs: customNewArgs);
+        if (PreMadeSolution == null)
+        {
+            Directory.CreateDirectory(projectDirectory);
+            dotNetHelper.ExecuteNew(Template.GetName(), projectName, projectDirectory, Language.ToCliName(), customArgs: customNewArgs);
+        }
+        else
+        {
+            string PreMadeName = PreMadeSolution.Split(Path.DirectorySeparatorChar).Last();
+            projectDirectory = Path.Combine(testRoot, PreMadeName);
+            string fullPreMadePath = Path.Combine(
+                dotNetHelper.FindRepoRoot(Environment.CurrentDirectory, false),
+                PreMadeSolution);
+            dotNetHelper.CopyHelper(projectDirectory, fullPreMadePath, true);
```

Comment:
We shouldn't be using the repo files as the source of the copy. Instead, they should be copied to the artifact layout as part of building the scenario tests repo.

Reply:
We can do this in the project file with `<Content` item groups: https://stackoverflow.com/questions/1292351/including-content-files-in-csproj-that-are-outside-the-project-cone

Reply:
Gotcha, I'll work on changing this. Definitely sounds a lot better than recursivly going up to find the repo root.

Reply:
Yep. You still have the right approach by copying to the target dir, but we can do it based on the output layout. I think you can get it laid out alongside the test dll, so you can find its directory, then look for the test resources there.

Reply:
You can see how we did this with the version identifier tests here https://github.com/dotnet/arcade/blob/main/src/Microsoft.DotNet.VersionTools/tests/Microsoft.DotNet.VersionTools.Tests.csproj#L17-L18

Reply:
I checked in a fix for this. I have a broad <None Include> statement in the csproj. The files are copied to output directory on main project build followed by being copied to the test directory during the test execution.